qiime/core/testing/data_layout.py

```diff
@@ -0,0 +1,131 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import collections
+import os
+import os.path
+
+###############################################################################
+#
+# int-sequence:
+#
+#     A sequence of integers stored in a single file. Since this is a sequence,
+#     the integers have an order and repetition of elements is allowed.
+#
+###############################################################################
+
+
+def int_sequence_validator(data_dir):
+    raise NotImplementedError()
+
+
+def int_sequence_to_list(data_dir):
+    with open(os.path.join(data_dir, 'ints.txt'), 'r') as fh:
+        view = []
+        for line in fh:
+            view.append(int(line.rstrip('\n')))
+        return view
+
+
+def list_to_int_sequence(view, data_dir):
+    with open(os.path.join(data_dir, 'ints.txt'), 'w') as fh:
```

Comment:
Note that the current formats (here and in q2-types) should be cross-platform compatible b/c universal newline mode is enabled by default, so `\n` can be safely used when reading/writing.